src/Sulu/Bundle/PersistenceBundle/DependencyInjection/PersistenceExtensionTrait.php

```diff
@@ -83,6 +83,14 @@ private function getRepositoryDefinition($object, array $services, ContainerBuil
             $this->getClassMetadataDefinition($services['model']),
         ]);
 
+        $repositoryReflectionClass = new \ReflectionClass($repositoryClass);
+        if ($repositoryReflectionClass->hasMethod('setAccessControlQueryEnhancer')) {
```

Comment:
Shouldn't make any difference, because it is very unlikely that this repository would implement the `setAccessControlQueryEnhancer` method then. However, still added the check, just to be sure.